### PR TITLE
bump mmcv version to 1.3.14 and add support for VFNet CPU

### PR DESCRIPTION
**Pull request by eugene123tw**
## Changes
* Bump `mmcv` to 1.3.14
* Add test of VFNet PyTorch CPU inference
* Add VFNet OpenVINO model export test

## Build number
* 267

### CODE REVIEW

File: mmdet/ops/nms/nms_wrapper.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def forward(ctx, bboxes, scores, iou_threshold, score_threshold, max_num, offset
             valid_mask = scores > score_threshold
         bboxes, scores = bboxes[valid_mask], scores[valid_mask]
         from mmcv.ops.nms import NMSop
-        inds = NMSop.forward(ctx, bboxes, scores, iou_threshold, offset)
+        inds = NMSop.forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold, max_num)
         inds = inds[:max_num]
```
Comment (druzhkov-paul) [inline mmdet/ops/nms/nms_wrapper.py:17]: Does all of this extra logic of any use now?

@SemyonBevzuk Could you suggest something here? No need to investigate it, just asking if you know it already.

File: tests/test_ote_inference.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        prog_bar = mmcv.ProgressBar(len(dataset))
+        for data in data_loader:
+            # just get the actual data from DataContainer
+            data['img_metas'] = data['img_metas'][0].data
+            data['img'] = [data['img'][0].data]
+            if next(model.parameters()).is_cuda:
+                # scatter to specified GPU
+                data = scatter(data, [device])[0]
+            with torch.no_grad():
+                result = model(return_loss=False, rescale=True, **data)
```
Comment (druzhkov-paul) [inline tests/test_ote_inference.py]: Does it work on CPU?

File: tests/test_ote_inference.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            batch_size = len(result)
+            results.extend(result)
+            for _ in range(batch_size):
+                prog_bar.update()
+        kwargs = config.get('evaluation', {})
+        kwargs.pop('interval', None)
+        kwargs.pop('gpu_collect', None)
```
Comment (druzhkov-paul) [inline tests/test_ote_inference.py]: What is it for?

File: tools/test_exported.py [hunk 4]
```diff
@@ -160,7 +161,12 @@ def main(args):
     results = []
     prog_bar = mmcv.ProgressBar(len(dataset))
     for i, data in enumerate(data_loader):
-        im_data = data['img'][0].cpu().numpy()
+        if torch.is_tensor(data['img'][0]):
+            im_data = data['img'][0].cpu().numpy()
+        elif isinstance(data['img'][0], DataContainer):
+            im_data = data['img'][0].data[0].cpu().numpy()
+        else:
+            raise RuntimeError('Unknown input')
```
Comment (druzhkov-paul) [inline tools/test_exported.py]: What is this?

File: mmdet/ops/nms/nms_wrapper.py (hunk 1 above)
Comment (SemyonBevzuk) [reply mmdet/ops/nms/nms_wrapper.py:17]: @druzhkov-paul
This [mmcv::NMSop](https://github.com/open-mmlab/mmcv/blob/be5841e45d42d7ffe9ee5e29d6348a84fe141caa/mmcv/ops/nms.py#L16-L33) is used here.
Therefore, there may be problems with NNCF, because `valid_mask` is created with NNCF tracing disabled.
In the implementation of mmcv::NMSop, with the non-zero `score_threshold`, a similar mask will be created, but with NNCF tracing enabled.

The `max_num` parameter can be passed, then `inds = inds[:max_num]` can be removed from the current NMSop.

Comment (druzhkov-paul) [reply mmdet/ops/nms/nms_wrapper.py:17]: Thanks @SemyonBevzuk .

Then, I'd propose to leave it as is for now.

File: tools/test_exported.py (hunk 4 above)
Comment (eugene123tw) [reply tools/test_exported.py]: I've found a bug running this script. `mmcv` changed the way how dataset is wrapped in DataContainer for CPU mode.

File: tests/test_ote_inference.py (hunk 3 above)
Comment (eugene123tw) [reply tests/test_ote_inference.py]: removed

File: tests/test_ote_inference.py (hunk 2 above)
Comment (eugene123tw) [reply tests/test_ote_inference.py]: Hi Pavel, yes it does. Please check the test results in build 267 for this specific test case: `test_ote_inference.py::TestInference::test_inference`

File: tools/test_exported.py [hunk 5]
```diff
@@ -160,7 +161,7 @@ def main(args):
     results = []
     prog_bar = mmcv.ProgressBar(len(dataset))
     for i, data in enumerate(data_loader):
-        im_data = data['img'][0].cpu().numpy()
+        im_data = data['img'][0].data[0].cpu().numpy()
```
Comment (eugene123tw) [inline tools/test_exported.py:165]: @druzhkov-paul I've found a bug running this script. `mmcv` changed the way how a dataset is wrapped in DataContainer for CPU mode.

File: tests/test_ote_inference.py (hunk 2 above)
Comment (druzhkov-paul) [reply tests/test_ote_inference.py]: Cool, thanks. One thing that I'm a bit concerned here is that we never really do calls to the model itself, but only to its wrapped instance. See here https://github.com/openvinotoolkit/mmdetection/blob/dp/nous_sdk_api/mmdet/apis/ote/apis/detection/inference_task.py#L223-L227 . So, I'm not sure that testing one guarantees that the other also works fine. What do you think?

File: tools/test_exported.py [hunk 6]
```diff
@@ -17,7 +17,8 @@
 import mmcv
 import numpy as np
 import sys
-from mmcv.parallel import collate
+import torch
+from mmcv.parallel import DataContainer, collate
```
Comment (druzhkov-paul) [inline tools/test_exported.py]: `DataContainer` seems to be unused now.

File: tests/test_ote_inference.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+            dist=False,
+            shuffle=False)
+
+        if 'cuda' in device and torch.cuda.is_available():
+            device_id = int(device.split(':')[-1])
+            model = MMDataParallel(
+                model.cuda(device_id), device_ids=[device_id])
+        else:
+            model = MMDataCPU(model)
+            model.to(device)
```
Comment (eugene123tw) [inline tests/test_ote_inference.py:126]: @druzhkov-paul I think either way should be fine, but just to be safe I've changed this using `MMDataParallel` and `MMDataCPU` to keep consistency running inference. I rerun the build and it works fine, please check Build 308

File: tests/test_ote_inference.py (hunk 2 above)
Comment (eugene123tw) [reply tests/test_ote_inference.py]: I think either way should be fine, but just to be safe I've changed this using MMDataParallel and MMDataCPU to keep consistency running inference. I rerun the build and it works fine, please check Build 308

File: tools/test_exported.py (hunk 6 above)
Comment (eugene123tw) [reply tools/test_exported.py]: 1.  I think either way should be fine, but just to be safe I've changed this using MMDataParallel and MMDataCPU to keep consistency running inference. I rerun the build and it works fine, please check Build 308
2. Unused imports removed

**Review by druzhkov-paul — APPROVED** [review]
